### ISSUE TITLE
Type export to FORTE not working except in top-level path of Type Library

### ISSUE THREAD

**Issue by dok-net** [eclipse-4diac/4diac-ide#53]
The dialog for exporting function blocks to FORTE source code stalls on any sub-folder of Type Library, no matter if the type is a stock FB or and own one - I am using symbolic links into another Git repository outside the 4diac Eclipse workspace, if that matters.

Workaround: copying the types into Type Library top-level and exporting from there allows the export dialog to complete.

**Comment by ernstblechaPT** [comment]
Yes, the symbolic links might matter. Are you using the symbolic links provided by your operating system (If yes: what operating system are you using?) or are you using the file-linking infrastructure within eclipse?

**Comment by kaartee** [comment]
I am using only what the Eclipse interface offers, running 4diac on Windows 10 and 11, anyway.
Even stock types shipping with the 4diac IDE cannot be exported to FORTE C++ source code, the "Finish" button in the export dialog remains grayed out.

**Comment by kaartee** [comment]
This applies to current (as of yesterday) builds milestone and nightly (4diac-ide_3.0.0-win32.win32.x86_64.zip, 2024-04-12).

**Comment by ernstblechaPT** [comment]
Do you also get the "No type selected!" errormessage?
![image](https://github.com/eclipse-4diac/4diac-ide/assets/142894096/ff17a01e-acc0-47a8-97cd-80027af336d3)

If yes: it seems that the export dialog does not support selecting linked files

**Comment by kaartee** [comment]
Rename is also broken. Copy & paste as well.

**Comment by kaartee** [comment]
> Do you also get the "No type selected!" errormessage?

Yes.

**Comment by kaartee** [comment]
@ernstblechaPT Copy and paste of an FB to clone one for modification to create further FBs doesn't work quite right, as well. Only the symbolic tree entry gets duplicated, the underlying file is not, but the FB name is modified, leaving the original FB link broken due to the name changing. This may cause some headache if going unnoticed for a while of incremental changes.

**Comment by oberlehner** [comment]
> Rename is also broken. Copy & paste as well.

What exactly are you renaming ?

**Comment by kaartee** [comment]
Given some Virtual Folder in the Type Library folder of a project contains an FB as a Linked File. Renaming that FB resource changes the Virtual Folder's entry, as well as the `<FBType Name="..."` attribute inside that file. But it does not adapt the file's name itself. It doesn't even mention it or offer to do so case by case. For copy and paste, the paste step implies renaming, and the same thing happens, or rather doesn't happen.
Perhaps I shouldn't use virtual folders. But I do have a need to version control my FBs.

**Comment by ernstblechaPT** [comment]
> Do you also get the "No type selected!" errormessage?
> 
> If yes: it seems that the export dialog does not support selecting linked files

Took another look into this: It appears that eclipse-platform took the choice that export will ignore linked files. Which is a somewhat sensible choice when working with java - you may want to link in some library files; but during export you want to ignore those files as they are anyway mentioned in the .project file which is "exported" when you use the "Archive File" or "File System" exporter from "General"

**Comment by kaartee** [comment]
Has this issue been fixed? While the export dialog still doesn't list the indiviidual FBs for export, the very button is active and the following steps, like generating C++ code, opening the merge tabs and asking the user for instructions, all appear working now.

**Comment by azoitl** [comment]
This has been fixed in such a way that standard and external libs are not located in the type lib folders anymore. You can only export the types from your own project.
